hardware/gpt-utils/gpt-utils.cpp

```diff
@@ -1072,12 +1072,14 @@ static int get_dev_path_from_partition_name(const char *partname,
                 if (stat(path, &st)) {
                         goto error;
                 }
-                if (readlink(path, buf, buflen) < 0)
-                {
+                char resolved[PATH_MAX] = {0};
```

Comment:
Is there a particular reason to use a separate buffer on the stack, and copy back to `buf` after the fact? The original code used `buf` directly as well.
When encountering an error and returning it, any caller is hopefully supposed to treat `buf` as containing undefined data.

Reply:
`buf`has no guarantees regarding its length. at the same time, `realpath` expects PATH_MAX sized string. So, this is to avoid some odd out-of-memory error induced if `buf` is shorter.

So far, the calling code had `buf` length set to PATH_MAX, but I haven't checked it systematically. and since API doesn't force it, we cannot assume `buf` length either.

Reply:
That is true, unfortunately `realpath` expects the buffer to be at least `PATH_MAX` bytes in size. Though returning an error when it's smaller - even when `PATH_MAX` is too much to realistically fit the string - might not be desirable. For correctness it should return an error when `buflen < pos + 2` before doing the `strncopy`, though.

Reply:
Thanks, I will add it. Will probably submit it tomorrow - would like to run the corrected code on device as well